picard/cluster.py

```diff
@@ -486,22 +521,33 @@ def get_cluster_title(self, cluster):
 
         return maxWord
 
-    def cluster(self, threshold):
-
+    def cluster(self, threshold, tagger=None):
         # Keep the matches sorted in a heap
         heap = []
+        num_files = self.cluster_dict.get_size()
+
+        # 20 evenly spaced indexes of files being clustered, used as checkpoints for every 5% progress
+        status_update_steps = set([int(i*num_files/20) for i in range(1, 21)])
```

Comment:
Since it is used twice in same file, make it a small function ?
Something like:

```python
def progress_checkpoints(num_files, each_n_percent=10):
    """"Create a set of evenly spaced indexes of files being clustered, used as checkpoints for progress""""
    num_checkpoints = int(100/each_n_percent)
    return set([int(i*num_files/num_checkpoints) for i in range(1, num_checkpoints+1)])


    status_update_steps = progress_checkpoints(num_files, each_n_percent=5)
```

Or even a small class:

```python
class ProgressCheckpoints:

    def __init__(self, num_files, each_n_percent=10):
        """"Create a set of evenly spaced indexes of files being clustered, used as checkpoints for progress""""
        num_checkpoints = int(100/each_n_percent)
        self._checkpoints = set([int(i*num_files/num_checkpoints) for i in range(1, num_checkpoints+1)])

    def is_checkpoint(self, index):
      return index in self._checkpoints

```

```python
status_update_steps = ProgressCheckpoints(num_files, each_n_percent=20)

if status_update_steps.is_checkpoint(y): ...

```

It can then moved to utils/ or smt to be reused elsewhere

Reply:
OK

Reply:
Ended up improving the logic. :D